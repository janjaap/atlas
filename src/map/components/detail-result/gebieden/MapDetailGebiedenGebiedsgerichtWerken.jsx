import React from 'react';
import PropTypes from 'prop-types';

import MapDetailResultWrapper from '../MapDetailResultWrapper';
import MapDetailResultItem from '../MapDetailResultItem';

const MapDetailGebiedenGebiedsgerichtWerken = ({
  panoUrl, gebiedsgerichtWerken, onMaximize, onPanoPreviewClick
}) => (
  <MapDetailResultWrapper
    panoUrl={panoUrl}
    onMaximize={onMaximize}
    onPanoPreviewClick={onPanoPreviewClick}
    subTitle={gebiedsgerichtWerken.label}
    title="Gebiedsgerichtwerken-gebied"
  >
    <ul className="map-detail-result__list">
      <MapDetailResultItem
        label="Code"
        value={gebiedsgerichtWerken.code}
      />
    </ul>
  </MapDetailResultWrapper>
);

MapDetailGebiedenGebiedsgerichtWerken.propTypes = {
  gebiedsgerichtWerken: PropTypes.shape({
    label: PropTypes.string.isRequired,
    code: PropTypes.string
  }).isRequired,
  panoUrl: PropTypes.string.isRequired,
  onMaximize: PropTypes.func.isRequired,
  onPanoPreviewClick: PropTypes.func.isRequired
};

export default MapDetailGebiedenGebiedsgerichtWerken;
