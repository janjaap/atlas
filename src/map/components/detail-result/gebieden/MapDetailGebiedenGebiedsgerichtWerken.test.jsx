import React from 'react';
import { shallow } from 'enzyme';

import MapDetailGebiedenGebiedsgerichtWerken from './MapDetailGebiedenGebiedsgerichtWerken';

describe('MapDetailGebiedenGebiedsgerichtWerken', () => {
  it('should render everything', () => {
    const gebiedsgerichtWerken = {
      label: 'Gebiedsgericht werken label',
      code: 'A'
    };
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailGebiedenGebiedsgerichtWerken
        panoUrl="panoUrl"
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
        gebiedsgerichtWerken={gebiedsgerichtWerken}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
