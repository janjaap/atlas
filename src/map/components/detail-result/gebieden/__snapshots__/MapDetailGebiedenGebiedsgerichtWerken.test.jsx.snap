// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapDetailGebiedenGebiedsgerichtWerken should render everything 1`] = `
<MapDetailResultWrapper
  onMaximize={[MockFunction]}
  onPanoPreviewClick={[MockFunction]}
  panoUrl="panoUrl"
  subTitle="Gebiedsgericht werken label"
  title="Gebiedsgerichtwerken-gebied"
>
  <ul
    className="map-detail-result__list"
  >
    <MapDetailResultItem
      label="Code"
      value="A"
    />
  </ul>
</MapDetailResultWrapper>
`;
